src/Stage/Parse/Parser.elm

```diff
@@ -548,9 +548,21 @@ expr =
                             }
                     )
                 )
-            , PP.infixLeft 1 (P.symbol (P.Token "++" ExpectingConcatOperator)) (Located.merge ListConcat)
-            , PP.infixLeft 1 (P.symbol (P.Token "+" ExpectingPlusOperator)) (Located.merge Plus)
-            , PP.infixRight 1 (P.symbol (P.Token "::" ExpectingConsOperator)) (Located.merge Cons)
+            , PP.infixLeft 1
+                (checkIndent (<) ExpectingIndentation
+                    |> P.andThen (\() -> P.symbol (P.Token "++" ExpectingConcatOperator))
+                )
+                (Located.merge ListConcat)
+            , PP.infixLeft 1
+                (checkIndent (<) ExpectingIndentation
+                    |> P.andThen (\() -> P.symbol (P.Token "+" ExpectingPlusOperator))
+                )
+                (Located.merge Plus)
+            , PP.infixRight 1
+                (checkIndent (<) ExpectingIndentation
+                    |> P.andThen (\() -> P.symbol (P.Token "::" ExpectingConsOperator))
+                )
+                (Located.merge Cons)
```

Comment:
@Janiczek this new code is _more correct_ but less readable and less DRY. Do you have any thoughts on how it can be tidied?

Reply:
Soon we'll remove these three cases and replace them with a generic operator. So there will be only one case. Does that sound reasonable?

Reply:
Sounds reasonable! I guess this applies to my colment below too!